components/NavBar/index.js

```diff
@@ -1,68 +1,224 @@
-import Link from 'next/link';
-import React, { useState, useRef } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
+import { useRouter } from 'next/router';
 import styles from './navbar.module.css';
+import Link from 'next/link';
 import Image from 'next/image';
 import GenericClosePopUp from '../Close-popup/GenericClosePopUp';
+import { USER_DATA_URL, LOGIN_URL, PATHS } from 'constants.js';
 
-const NavBar = ({ personData: { photo } }) => {
-  const RDSLogo = '/assets/Real-Dev-Squad1x.png';
+const NavBar = () => {
+  const router = useRouter();
+
+  const RDS_LOGO = '/assets/Real-Dev-Squad1x.png';
+  const GITHUB_LOGO = '/assets/github.png';
+  const [userData, setUserData] = useState({});
   const [toggle, setToggle] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [mountedComponent, setMountedComponent] = useState(false);
   const navbarRef = useRef();
   GenericClosePopUp(navbarRef, () => {
     setToggle(false);
   });
 
-  return (
-    <nav className={styles.navBar}>
-      <ul>
-        <li>
-          <div className={styles.logo}>
-            <Link href="https://realdevsquad.com">
-              <Image src={RDSLogo} alt="Navbar Logo" width="50" height="50" />
-            </Link>
-          </div>
-        </li>
-        {/* <li>
-          <Link href="/shop">
-            <a>Shop</a>
-          </Link>
-        </li>
-        <li>
-          <Link href="/cart">
-            <a>Cart</a>
-          </Link>
-        </li> */}
-        <li>
-          <Link href="/auction">
-            <a>Auction</a>
-          </Link>
-        </li>
-        <li>
-          <Link href="/trading">
-            <a>Stocks</a>
-          </Link>
-        </li>
-      </ul>
+  useEffect(() => {
+    const fetchData = async () => {
+      await fetch(USER_DATA_URL, { credentials: 'include' })
+        .then((response) => response.json())
+        .then((responseJson) => {
+          if (responseJson.incompleteUserDetails) {
+            return window.location.replace(
+              'https://my.realdevsquad.com/signup'
+            );
+          }
+          responseJson.statusCode === 401
+            ? setIsLoggedIn(false)
+            : setIsLoggedIn(true);
+
+          setUserData({
+            userName: responseJson.username,
+            firstName: responseJson.first_name,
+          });
+        })
+        .catch((err) => {
+          console.error(err);
```

Comment:
Maybe show the error message to user, or do some else operation. No need for a console.

Reply:
Please suggest an error message and a way to show it to the user.
I think using an `alert()` method with 500 Internal Server Error message

Reply:
Alert with `Something went wrong!` would work.

Reply:
> Alert with `Something went wrong!` would work.

@shubham-y If you have any confusion about error message to show, then maybe take the help of other rds members.

Reply:
Using an `alert()` method to show the message does not seem correct. Instead, we can use the [react-toastify](https://www.npmjs.com/package/react-toastify) package to display an error message as done in the status site. We can create a new issue for the same.